.vscode/settings.json

```diff
@@ -1,3 +1,6 @@
 {
-	"go.buildTags": "e2e"
+	"files.associations": {
+		// test files embedded in e2e with a main method, can't be suffixed with .go because Go will complain about main
```

Comment:
I think we should move away from this embedding approach entirely, and move towards what we have in RP.

If we're breaking down the project into a series of cmds anyway, why not create a directory for each kind of deployment we create and containerize them? That way, like we do in rp, we can leverage things like syntax highlighting etc, and potentially move towards more complex clients/servers in the future without keeping them contained in single files.

Feel like that's broadly a more flexible approach anyway

Reply:
why not just do it here? this seems like the optimal time given that we're already rearranging their file structure, which is going to be the extent of the changes regardless

In other words - I'm not sure that this PR gets significantly more complex if we just make that change now vs leaving it as tech debt going forward